### ISSUE TITLE
Windows ico files with multiple images are downsampled or not displayed

### ISSUE THREAD

**Issue by biqqles** [Nuitka/Nuitka#977]
`--windows-icon-from-ico` works, but seems to downsample the source image to an unsightly degree.

The following screenshot shows the problem, with the icon on the right generated by PyInstaller from the exact same source image:

![image](https://user-images.githubusercontent.com/19939282/107259736-46e29f00-6a35-11eb-92b4-13f3e1da5d7d.png)

The original image can be found [here](https://raw.githubusercontent.com/biqqles/wingman/master/icons/general/wingman.ico). Thanks!

---

```
python -m nuitka --version
0.6.11.6
Python: 3.8.7 (tags/v3.8.7:6503f05, Dec 21 2020, 17:43:54) [MSC v.1928 32 bit (Intel)]
Executable: C:\Users\biqqles\Documents\Python\Python38-32\python.exe
OS: Windows
Arch: x86
```

**Comment by kayhayen** [comment]
Thanks for the example. Nuitka surely doesn't downsample, but Windows does. And Nuitka attempts to include all the icons, but maybe it fails in some way. I will use your example to find out what's going on there.

**Comment by kayhayen** [comment]
Just to clarify, you saw this icon for standalone or for onefile mode?

**Comment by kayhayen** [comment]
Or well, accelerated mode of course, although that should be the same as standalone mode.

**Comment by biqqles** [comment]
`--standalone`, yep.

**Comment by kayhayen** [comment]
I was trying it with accelerated, which also supports attaching icons. I didn't see any icon displayed, and when Visual Studio 2019 was used to open the exe, it said the icon resource was corrupt. That's of course a mess to debug. Hopefully comparing with what the alternative does will help to find out, where things go wrong. There are 5 icons in there, maybe also if I go one by one, it would be helpful. Adding a generally useless option to select the icon number I guess.

**Comment by kayhayen** [comment]
Adding a message what it is doing, and observing the icons, the first 3 icons work, but 4 and 5 do not work, and leave a black in my explorer window. With this new option `--windows-icon-from-ico=wingman.ico#4` I am getting this trace output and no icon visible:

> Nuitka-Postprocessing:INFO: Adding 1 icon(s) from icon file 'wingman.ico#4'.

**Comment by kayhayen** [comment]
Looking at the icon file in Visual Studio 2019 itself, no complaint about it, and the icons not working are those with sizes 64x64 and 128x128, so no wonder it would have to be blurry, if you had to go with the 48x48 one.

Also experimental computer science led me to copy the file by saving to another name, no change, delete the #1 icons, just to see if saving it again improves it, which it does not.

**Comment by biqqles** [comment]
Weird... but still, progress!

**Comment by kayhayen** [comment]
I figured it out, one of the header structure fields contains an id, and was oversized. It was the last field, so ids could be corrupted, and the icon group header got misaligned. The incorrect size is easy to correct. Expect a fix soon.

**Comment by kayhayen** [comment]
Ok, pushed it to factory https://nuitka.net/doc/factory.html as it works for me now, selecting any icon or all icons from your icon file just works. Didn't yet document the new option for selecting single icons, that's future work.

**Comment by biqqles** [comment]
Amazing, thank you!

**Comment by kayhayen** [comment]
This is released as 0.6.12.1 hotfix.
